internal/audit/audit.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"slices"
+
+	"github.com/google/uuid"
+)
+
+const (
+	AuditAdditionalDataGrantedPermissionKey = "authorization.permission"
+	AuditAdditionalDataGrantedBindingKey    = "authorization.binding"
+	AuditAdditionalDataGrantedRoleKey       = "authorization.role"
+)
+
+var reservedLabelKeys = []string{
+	AuditAdditionalDataGrantedPermissionKey,
+	AuditAdditionalDataGrantedBindingKey,
+	AuditAdditionalDataGrantedRoleKey,
+}
+
+type Audit struct {
```

Comment:
We should also set which service set the log? So, for which service rond is the sidecar?

Reply:
also the user agent and the IP. I've almost forgot about them